### PR TITLE
disttask: start dispatcher/scheduler of distatsk framework when tidb start

### LINKED ISSUE #42481: Start dispatcher and scheduler on each TiDB instance when node starts

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #42481

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by wjhuang2016 — APPROVED** [review]

File: disttask/framework/dispatcher/register.go [hunk 1]
```diff
@@ -37,6 +37,13 @@ func RegisterTaskFlowHandle(taskType string, dispatcherHandle TaskFlowHandle) {
 	taskFlowHandleMap.Unlock()
 }
 
+// ClearTaskFlowHandle is only used in test
+func ClearTaskFlowHandle() {
+	taskFlowHandleMap.Lock()
+	taskFlowHandleMap.handleMap = make(map[string]TaskFlowHandle)
```
Comment (hawkingrei) [inline disttask/framework/dispatcher/register.go:43]: ```maps.Clear(taskFlowHandleMap.handleMap)```

to avoid extra memory allocation.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**